internals/overlord/state/notices_test.go

```diff
@@ -176,30 +219,72 @@ func (s *noticesSuite) testRepeatAfter(c *C, first, second, delay time.Duration)
 	c.Assert(newLastRepeated.After(lastRepeated), Equals, true)
 }
 
+func (s *noticesSuite) TestNoticesFilterUserID(c *C) {
+	st := state.New(nil)
+	st.Lock()
+	defer st.Unlock()
+
+	uid1 := uint32(1000)
+	uid2 := uint32(0)
+	addNotice(c, st, &uid1, state.CustomNotice, "foo.com/bar", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, &uid2, state.ChangeUpdateNotice, "123", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, &uid2, state.WarningNotice, "Warning 1!", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, nil, state.WarningNotice, "Warning 2!", nil)
+
+	// No user IDs
```

Comment:
Should this say `// Nil filter arg`?

Reply:
I like `// User ID unset`, followed by `// User ID set`. I'm adjusting all of these a bit, let me know what you think.